internal/domain/usecases/attachment_usecase.go

```diff
@@ -123,5 +123,19 @@ func (u *attachmentUsecase) uploadAndSaveAttachments(fileReaders map[string]io.R
 }
 
 func (u *attachmentUsecase) DeleteAttachment(ID string) *apperror.AppError {
+	_, err := u.attachmentRepository.FindAttachmentByID(ID)
+	if err != nil {
+		u.logger.Named("DeleteAttachment").Error("Find attachment by ID: ", zap.Error(err))
+		return apperror.NotFoundError(fmt.Sprintf("attachment not found: %s", err.Error()))
+	}
```

Comment:
we can use this below pattern instead na

```go
	if _, err := u.attachmentRepository.FindAttachmentByID(ID); err != nil {
                if errors.Is(err, gorm.ErrRecordNotFound) {
                 // error not found
                }
		u.logger.Named("DeleteAttachment").Error("Find attachment by ID: ", zap.Error(err))
		return apperror.NotFoundError(fmt.Sprintf("attachment not found: %s", err.Error()))
	}
```

Reply:
```golang
func (u *userUsecase) DeleteUserByID(req *dtos.UserDTO, userID string) *apperror.AppError {
	role, err := utils.GetRole(req.Role)
	if err != nil {
		u.logger.Named("DeleteUserByID").Error(constant.ErrInvalidRole, zap.Error(err))
		return apperror.BadRequestError(constant.ErrInvalidRole)
	}

	existingUser, err := u.userRepository.FindUserByID(userID)
	if err != nil {
		u.logger.Named("DeleteUserByID").Error(constant.ErrUserNotFound, zap.String("userID", userID), zap.Error(err))
		return apperror.NotFoundError(constant.ErrUserNotFound)
	}

	if existingUser.RoleID != role {
		u.logger.Named("DeleteUserByID").Error(constant.ErrInvalidRole, zap.Error(err))
		return apperror.BadRequestError(constant.ErrInvalidRole)
	}

	err = u.userRepository.DeleteUserByID(userID)
	if err != nil {
		u.logger.Named("DeleteUserByID").Error(constant.ErrDeleteUserByID, zap.String("userID", userID), zap.Error(err))
		return apperror.InternalServerError(constant.ErrDeleteUserByID)
	}

	u.logger.Named("DeleteUserByID").Info("Success: ", zap.String("user_id", userID))
	return nil
}
```
Or like this? idk the correct instruction

Reply:
@wiraphatys

Reply:
up to you bro, its just minor lol

Reply:
nice my bro, seems like a non-serious reviewing. unlike my boss :(